nixops/backends/ec2.py

```diff
@@ -506,6 +506,8 @@ def create_after(self, resources, defn):
 
     def attach_volume(self, device, volume_id):
         volume = nixops.ec2_utils.get_volume_by_id(self.connect(), volume_id)
+        if not volume:
+            raise Exception("volume {0} doesn't exist".format(volume_id))
```

Comment:
Maybe you can suggest running `nixops check` here to reconcile the state with the real-world state.